bpfd/src/bpf.rs

```diff
@@ -875,66 +868,71 @@ impl BpfManager {
     // program is the map owner, then a new entry is add to the map hash
     // table and permissions on the directory are updated to grant bpfd
     // user group access to all the maps in the directory. If this eBPF
-    // program is not the owner, then the eBPF program UUID is added to
+    // program is not the owner, then the eBPF program ID is added to
     // the Used-By array.
     async fn save_map(
         &mut self,
-        id: Uuid,
-        map_owner_uuid: Option<Uuid>,
+        id: u32,
+        map_owner_id: Option<u32>,
         map_pin_path: &Path,
     ) -> Result<(), BpfdError> {
-        let (map_owner, _) = get_map_index(id, map_owner_uuid);
-
-        if map_owner {
-            let program = self
-                .programs
-                .get_mut(&id)
-                .expect("Program should be loaded");
-
-            let map = BpfMap { used_by: vec![id] };
-
-            self.maps.insert(id, map);
+        match map_owner_id {
+            Some(id) => {
```

Comment:
```suggestion
            Some(owner_id) => {
```
`id` already exists and needs to be used in this case statement. Change `id` to something else in this case statement.